aosp_diff/preliminary/external/drm_hwcomposer/0002-Fix-issue-have-OVERLAY-crtc-count-neq-connector-count.patch

```diff
@@ -0,0 +1,212 @@
+   for (auto &conn : connectors_) {
+@@ -281,7 +283,7 @@ std::tuple<int, int> DrmDevice::Init(const char *path, int num_displays) {
+         displays_[num_displays] = num_displays;
+         found_primary = true;
+         ++num_displays;
+-      } else if (conn->display() < 0) {
++      } else if (conn->display() < 0 && num_displays < max_pipe_count) {
+         conn->set_display(num_displays);
+         displays_[num_displays] = num_displays;
+         ++num_displays;
+@@ -320,7 +322,14 @@ std::tuple<int, int> DrmDevice::Init(const char *path, int num_displays) {
+       break;
+     }
+ 
+-    planes_.emplace_back(std::move(plane));
++    bool enable = RTPropertyCheck::CheckMultiPlaneEnable();
++    if(!enable) {
++      if (DRM_PLANE_TYPE_PRIMARY == plane->type())
++        planes_.emplace_back(std::move(plane));
++    }
++    else {
++      planes_.emplace_back(std::move(plane));
++    }
+   }
+   drmModeFreePlaneResources(plane_res);
+   if (ret)
+@@ -332,7 +341,44 @@ std::tuple<int, int> DrmDevice::Init(const char *path, int num_displays) {
+     return std::make_tuple(ret, 0);
+   }
+ 
++#if 0
++  for (auto &conn : connectors_) {
++    ret = CreateDisplayPipe(conn.get());
++    if (ret) {
++      ALOGE("Failed CreateDisplayPipe %d with %d", conn->id(), ret);
++      return std::make_tuple(ret, 0);
++    }
++    if (!AttachWriteback(conn.get())) {
++      ALOGI("Display %d has writeback attach to it", conn->display());
++    }
++  }
++#else
++  for (auto conn : primary_candidates) {
++    if(max_pipe_count-- <= 0)
++      break;
++  
++    ret = CreateDisplayPipe(conn);
++    if (ret) {
++          ALOGE("Failed CreateDisplayPipe %d with %d", conn->id(), ret);
++          return std::make_tuple(ret, 0);
++    }
++    if (!AttachWriteback(conn)) {
++          ALOGI("Display %d has writeback attach to it", conn->display());
++    }
++  }
++
+   for (auto &conn : connectors_) {
++    auto it = std::find_if(std::begin(primary_candidates),
++                           std::end(primary_candidates),
++                           [&conn](const DrmConnector *conn_pri) {
++                             return conn->id() == conn_pri->id();
++                           });
++    if (it != std::end(primary_candidates)) {
++      continue;
++    }
++    if(max_pipe_count-- <= 0)
++      break;  
++  
+     ret = CreateDisplayPipe(conn.get());
+     if (ret) {
+       ALOGE("Failed CreateDisplayPipe %d with %d", conn->id(), ret);
+@@ -342,6 +388,9 @@ std::tuple<int, int> DrmDevice::Init(const char *path, int num_displays) {
+       ALOGI("Display %d has writeback attach to it", conn->display());
+     }
+   }
++  
++#endif
++
+   return std::make_tuple(ret, displays_.size());
+ }
+ 
+diff --git a/include/drmhwcomposer.h b/include/drmhwcomposer.h
+index 1007144..32cc3ff 100644
+--- a/include/drmhwcomposer.h
++++ b/include/drmhwcomposer.h
+@@ -158,6 +158,17 @@ struct DrmHwcLayer {
+   }
+ };
+ 
++class RTPropertyCheck {
```

Comment:
why we need add a class for that, we only need get one prop, one function is enough, even we can directly call property_get in DrmDevice::Init